packages/core/src/drawer/drawer.spec.ts

```diff
@@ -0,0 +1,155 @@
+import type { Drawer, DrawerAttachement } from './drawer.js';
+import { expect, fixture } from '@open-wc/testing';
+import { html } from 'lit';
+import './register.js';
+import { SinonSpy, spy, stub } from 'sinon';
+
+describe('sl-drawer', () => {
+  let el: Drawer;
+
+  beforeEach(async () => {
+    el = await fixture(html`<sl-drawer></sl-drawer>`);
+  });
+
+  it('should render correctly', () => {
+    expect(el).shadowDom.to.equalSnapshot();
+  });
+
+  // it('should have a dialog role', async () => {
+  //   const { role } = await a11ySnapshot({ selector: 'dialog' }) as any;
+
+  //   expect(role).to.equal('dialog');
+  // });
+
+  describe('positioning', () => {
+    it('should attach the drawer to the right by default', () => {
+      expect(el).to.have.attribute('attachment', 'right');
+    });
+  
+    ['right', 'left', 'top', 'bottom'].forEach(attachment => {
+      it(`should support ${attachment} attachment`, async () => {
+        el.attachment = attachment as DrawerAttachement;
+        await el.updateComplete;
+  
+        expect(el).to.have.attribute('attachment', attachment);
+      });
+    });
+  });
+
+  describe('opening and closing', () => {
+    it('should not show the dialog by default', () => {
+      expect(el.renderRoot.querySelector('dialog')).not.to.have.attribute('open');
+    });
+    
+    it('should open and close the drawer', async () => {
+      const dialog = el.renderRoot.querySelector('dialog')
+      el.showModal();
+      await el.updateComplete;
+      
+      expect(dialog).to.have.attribute('open');
+      expect(document.documentElement.style.overflow).to.equal('hidden');
+      
+      el.close();
+      expect(dialog).not.to.have.attribute('open');
+     
+      // dispatch the event ourselves, because waiting for it to come from the actual dialog is too unreliable
+      dialog?.dispatchEvent(new Event('close'));
+      expect(document.documentElement.style.overflow).to.equal('');
+
+    });
+
+    it('should not close the drawer when the cancel event is fired but close is disabled', async () => {
+      const dialog = el.renderRoot.querySelector('dialog');
+      const cancelEvent = new Event('cancel');
+      const cancelEventSpy = spy(cancelEvent,'preventDefault')
+
+      
+      el.disableClose = true;
+      el.showModal();
+      await el.updateComplete;
+      
+      expect(dialog).to.have.attribute('open');
+      
+      dialog?.dispatchEvent(cancelEvent);
+      
+      expect(cancelEventSpy).to.have.been.called;
+    });
+
+    it('should close the drawer when the cancel event is fired and close isn`t disabled', async () => {
+      const dialog = el.renderRoot.querySelector('dialog');
+      const cancelEvent = new Event('cancel');
+      const cancelEventSpy = spy(cancelEvent,'preventDefault')
```

Comment:
Sigh, I need to figure out how to enable eslint for `*.spec.ts` files; you're missing a semicolon here :)

Reply:
being too reliant on linting makes you a lazy developer 🫣